### ISSUE TITLE
Option to disable headlines

### ISSUE THREAD

**Issue by MrMooky** [GrahamCampbell/Laravel-Markdown#118]
I was not able to see an option to disable headlines with a # mark in front. We don't want our users to mess too much with the output on the page and would like to disable rendering of headlines.

Is that possible with your plugin?

**Comment by colinodell** [comment]
Yep it's possible!  You'll just need to replace the default `Environment` (or tweak it to render headlines as empty strings).

See #108 and https://commonmark.thephpleague.com/customization/environment/ for some pointers

**Comment by ja-heller** [comment]
@colinodell @GrahamCampbell 

Is there any chance you can maybe include some lines of code, to show how to override the `Environment ` the best way, when using your lib? Looking for a way to disable the headlines on the fly too (for some user inputs I would like to disable, for some certain not).

**Comment by colinodell** [comment]
One possible approach would be to duplicate [the core extension](https://github.com/thephpleague/commonmark/blob/0.19.1/src/Extension/CommonMarkCoreExtension.php) and simply remove lines 32 and 35.  That will prevent the two types of headers from being parsed.  Alternatively, you could configure the environment to use a custom block renderer instead of [the default one](https://github.com/thephpleague/commonmark/blob/0.19.1/src/Block/Renderer/HeadingRenderer.php) to render the parsed headers as plain text instead of header tags.  I think the latter option might be easier to implement.

**Comment by ja-heller** [comment]
@colinodell 

What's the best place to place this code / how to override the code (`register`) without editing vendor based files?

**Comment by colinodell** [comment]
Here's a really easy solution:

**Step 1: Create a custom extension which uses the `ParagraphRenderer` to handle headings:**

```php
// TODO: Double-check these namespaces, I'm doing this from memory here :)
use League\CommonMark\Block\Element\Heading;
use League\CommonMark\Block\Renderer\ParagraphRenderer;
use League\CommonMark\ConfigurableEnvironmentInterface;
use League\CommonMark\Extension\ExtensionInterface;

final class DisableHeaderRenderingExtension implement ExtensionInterface
{
    public function register(ConfigurableEnvironmentInterface $environment)
    {
        $environment->addBlockRenderer(Heading::class, new ParagraphRenderer(), 999);
    }
}
```

By using a higher priority (`999`) than the default one (`0`), your renderer will be called instead of the default one.  The paragraph renderer works just like the heading one, except it'll use a `<p>` tag instead of `<h1>`.

**Step 2: Register this custom extension**

[Use the `extensions` option](https://github.com/GrahamCampbell/Laravel-Markdown#commonmark-extensions) to add your extension to Laravel Markdown.

Hope that helps!

**Comment by ja-heller** [comment]
@colinodell 

Thanks, for sure it will.
Where's the best place to place the custom `ParagraphRenderer`extension?

**Comment by colinodell** [comment]
Anywhere in your project that you think makes sense should be fine.
